lib/app/views/guided.erb

```diff
@@ -158,77 +158,92 @@
-                      </button>
-                    <% when 'cancel' %>
-                      <div class="group-message">
-                        Cancelled
-                      </div>
-                      <%= markdown_to_html(group.overview) %>
-                      <button class='sequence-button btn btn-info btn-sm' data-group-id='<%=group.id%>'>
-                        <span class="oi oi-media-play"></span> Begin Tests
-                      </button>
-                    <% when 'error' %>
-                      <div class="group-message">
-                        Errored
-                      </div>
-                      <%= markdown_to_html(group.overview) %>
-                      <% if group.run_skipped && group.test_cases.any?{ |test_case| sequence_results[test_case.id].try(:result) == 'skip'} %>
-                        <button class='sequence-button btn btn-info btn-sm' data-group-id='<%=group.id%>' data-skipped-only='true'>
-                          <span class="oi oi-media-play"></span> Rerun Skipped Tests
+                  </div>
+                  <%= markdown_to_html(group.overview) %>
+
+                  <div class='buttons-container'>
+                    <% case group_result.to_s
+                      when 'pass' %>
+                        <button class='sequence-button btn btn-info btn-sm' data-group-id='<%=group.id%>'>
+                          <span class="oi oi-loop"></span> Rerun All Tests
                         </button>
-                      <% end %>
-                      <button class='sequence-button btn btn-info btn-sm' data-group-id='<%=group.id%>'>
-                        <span class="oi oi-media-play"></span> Rerun All Tests
-                      </button>
-                    <% when 'wait' %>
-                      <div class="group-message">
-                        Paused
-                      </div>
-                      <%= markdown_to_html(group.overview) %>
-                      <button class='sequence-button btn btn-info btn-sm' data-group-id='<%=group.id%>'>
-                        <span class="oi oi-media-play"></span> Begin Tests
-                      </button>
-                    <% when 'skip' %>
-                        <div class="group-message">
-                        Skipped
-                      </div>
-                      <%= markdown_to_html(group.overview) %>
-                      <% if group.run_skipped && group.test_cases.any?{ |test_case| sequence_results[test_case.id].try(:result) == 'skip'} %>
-                        <button class='sequence-button btn btn-info btn-sm' data-group-id='<%=group.id%>' data-skipped-only='true'>
-                          <span class="oi oi-media-play"></span> Rerun Skipped Tests
+                      <% when 'fail' %>
+                        <% if group.run_skipped && group.test_cases.any?{ |test_case| sequence_results[test_case.id].try(:result) == 'skip'} %>
```

Comment:
Assuming that `group` is a `TestGroup`, this condition looks like it could be a `TestGroup` instance method. Then this and the ifs could just be something like `if group.skipped?`

Reply:
The group doesn't have the results attached to it, it is just the metadata that defines the groups.  But I'll clean this up a little.